code/obj/item/stun_baton.dm

```diff
@@ -419,6 +419,50 @@
 		logTheThing("combat", user, null, "swaps the power cell (<b>Cell type:</b> <i>[C.type]</i>) of [src] at [log_loc(user)].")
 		return
 
+	intent_switch_trigger(var/mob/user)
+		src.do_flip_stuff(user, user.a_intent)
+
+	attack_hand(var/mob/user)
+		if (src.flipped && user.a_intent != INTENT_HARM)
+			user.show_text("You flip \the [src] the right way around as you grab it.")
+			src.flipped = false
+			src.update_icon()
+			user.update_inhands()
+		else if (user.a_intent == INTENT_HARM)
+			src.do_flip_stuff(user, INTENT_HARM)
+		..()
+
+	proc/do_flip_stuff(var/mob/user, var/intent)
+		if (intent == INTENT_HARM)
+			if (src.flipped) //swapping hands triggers the intent switch too, so we dont wanna spam that
+				return
+			src.flipped = true
+			animate(src, transform = turn(matrix(), 120), time = 0.07 SECONDS) //turn partially
+			animate(transform = turn(matrix(), 240), time = 0.07 SECONDS) //turn the rest of the way
+			animate(transform = turn(matrix(), 180), time = 0.04 SECONDS) //finish up at the right spot
+			src.transform = null //clear it before updating icon
+			src.update_icon()
+			user.update_inhands()
+			user.show_text("<B>You flip \the [src] and grab it by the head! [src.status ? "It seems pretty unsafe to hold it like this while it's on!" : "At least its off!"]</B>", "red")
+		else //not already flipped
+			if (!src.flipped) //swapping hands triggers the intent switch too, so we dont wanna spam that
+				return
+			src.flipped = false
+			animate(src, transform = turn(matrix(), 120), time = 0.07 SECONDS) //turn partially
+			animate(transform = turn(matrix(), 240), time = 0.07 SECONDS) //turn the rest of the way
+			animate(transform = turn(matrix(), 180), time = 0.04 SECONDS) //finish up at the right spot
+			src.transform = null //clear it before updating icon
+			src.update_icon()
+			user.update_inhands()
+			user.show_text("<B>You flip \the [src] and grab it by the base!", "red")
+
+	dropped(mob/user)
+		if (src.flipped)
+			src.flipped = false
+			src.update_icon()
+			user.update_inhands()
+			..()
```

Comment:
why are you calling them at this indentation level

Reply:
oops youre right, thats dumb